lib/ProductOpener/Food.pm

```diff
@@ -1345,6 +1538,10 @@ sub compute_nutrition_score ($product_ref) {
 
 		foreach my $nid ("energy", "fat", "saturated-fat", "sugars", "sodium", "proteins") {
 			if (not defined $product_ref->{nutriments}{$nid . $prepared . "_100g"}) {
+				# we have two special case where we can deduce data
+				next
+					if (saturated_fat_0_because_of_fat_0($product_ref, $prepared)
+					|| sugar_0_because_of_carbohydrates_0($product_ref, $prepared));
```

Comment:
I think this will fail if you don't have both saturated_fat and sugars, but you have 1 of carbohydrates == 0 or fat == 0. You will get a Nutri-Score even though you can deduce only one of them.

Reply:
Is that ok @stephanegigandet (I added a test, so that should do)